secrets/azurekeyvault/akv.go

```diff
@@ -254,6 +254,9 @@ func (k *keeper) getKeyVaultURI() string {
 }
 
 func (k *keeper) validateOptions() error {
+	if k.options == nil {
```

Comment:
How about we remove `validateOptions` and put the code in the constructor? Giving the error at construction time is better than giving it at `Encrypt/Decrypt` time, no?